osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/config/kafka/AbstractKafkaConsumerConfig.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ */
+package org.opensmartgridplatform.shared.application.config.kafka;
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.Environment;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+
+public abstract class AbstractKafkaConsumerConfig<K, V> extends KafkaConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractKafkaConsumerConfig.class);
+
+    private ConsumerFactory<K, V> consumerFactory;
+    private ConcurrentKafkaListenerContainerFactory<K, V> kafkaListenerContainerFactory;
+
+    public AbstractKafkaConsumerConfig(final Environment environment, final String propertiesPrefix, final String topic,
+            final int concurrency, final int pollTimeout) {
+        super(environment);
+
+        LOGGER.debug("=================================================================================");
+        LOGGER.info("Initializing Kafka Consumer for Topic {}", topic);
+        LOGGER.debug("Common properties prefix: {}", propertiesPrefix);
+        LOGGER.debug("Consumer properties prefix: {}", getConsumerPropertiesPrefix(propertiesPrefix));
+        LOGGER.debug("=================================================================================");
+
+        this.initKafkaConsumer(propertiesPrefix, topic, concurrency, pollTimeout);
+    }
+
+    public abstract ConsumerFactory<K, V> consumerFactory();
+
+    public abstract ConcurrentKafkaListenerContainerFactory<K, V> kafkaListenerContainerFactory();
+
+    protected ConsumerFactory<K, V> getConsumerFactory() {
+        return this.consumerFactory;
+    }
+
+    protected ConcurrentKafkaListenerContainerFactory<K, V> getKafkaListenerContainerFactory() {
+        return this.kafkaListenerContainerFactory;
+    }
+
+    private static String getConsumerPropertiesPrefix(final String propertiesPrefix) {
+        return propertiesPrefix + ".consumer";
+    }
+
+    private void initKafkaConsumer(final String propertiesPrefix, final String topic, final int concurrency,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "topic". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1172)